database/migrations/2023_11_22_121149_create_values_table.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration {
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('attributes', function (Blueprint $table) {
+            $table->dropColumn('model');
+        });
+
+        Schema::create('values', function (Blueprint $table) {
+            $table->char('id', 36)->collation('ascii_bin')->primary();
+
+            $table->char('attribute_id', 36)->collation('ascii_bin');
+            $table->char('object_id', 36)->collation('ascii_bin');
+
+            $table->char('position_id', 36)->collation('ascii_bin')->nullable();
+            $table->char('ref_object_id', 36)->collation('ascii_bin')->nullable();
```

Comment:
model_object_id albo model_value_id może. Żeby te kolumny z wartością z których dokładnie jedna jest ustawiona miały value w nazwie. To jeszcze wtedy position_value_id by pasowało. W każdym razie myślę że to by poprawiło zrozumiałość tej tabeli dla kogoś kto na nią spojrzy nie znając dobrze całego tego systemu atrybutów. Bo jak sądzę nie ma nigdzie słownej dokumentacji poszczególnych kolumn, tylko plik md (który może zaktualizujesz?), również bez opisu słownego. Ja to w ogóle uważam że słowne opisy to przydatna rzecz w różnych miejscach, i gdzieś w tej tabeli by się przydała, nie wiem gdzie to się umieszcza. No chyba że jest taka konwencja żeby tabele nie były same z siebie zrozumiałe :)

Reply:
no pliku tables/values.md brakuje, a plik tables/attribures.md się trochę zdezaktualizował, dodam kiedyś niedługo

Reply:
Ja sie zgodzę z Tomkiem tu, że nie jest zbyt zrozumiałe na co to są referencje w tych kolumnach - mówiłem to już przy dodawaniu pierwszej tabeli z x_dict_id - to wygląda jak ref na dictionary, nie pozycję w nim.

Dokumentacja pomaga, choć wciąż z niej samej (bez czytania migracji i obczajania o co chodzi w danych nie wiem co to object_id i ref_object_id.

Reply:
pierwotnie tu było nie ref_dict_id tylko position_id, co nie było szczególnie dobre, bo {table}_id raczej to jest relacja, że to coś jest jakoś nadrzędnem tutaj to nie pasuje

dict_id jako position_id, ale zawsze w jakimś stopniu w kontekście konkretnego dictionary, wydaje mi się najlepszą z opcji w tym temacie. No i po wymyśleniu tego nie byłem pewien czy to jest ok, ale jestem przekoany że jest, mi się z tym dobrze pracuje, mimo że co raz ktoś narzeka, że głupie

no to, że obiekt value może mieć 2 referencje do dwóch praktycznie dowolnych obiektów, no to musi mieć jakąś ogólną nazwę, z propozycji:
- value_id  - brzmi jak id z obiektu Value
- value_object_id - brzmi jakby tu były jakieś value objecty
- object_id_value - to _id się trochę ukrywa w środku
- id_value - może by mogło być, ale jako ąże chciałem oddzielną kolumnę na id ze słownika, to same brzydkie nazwy z tego wychodzą

Reply:
Słowo object nie byłoby tu chyba dobre bo object oznacza to na czym jest ten atrybut i to by wprowadzało zamieszanie.

Więc id_value wydaje się rozsądne. Czy jest jakiś dobry powód żeby trzymać osobno id pozycji słownikowych i osobno id modeli? I tak trzeba zajrzeć do opisu atrybutu żeby skutecznie zrozumieć tę wartość.

Reply:
id pozycji słownika może mieć klucz obcy i nie przyjmować śmieci, a id_czegokowiek nie może

a najcześniej to będzie od ze słownika, więc warto, żeby zazwyczaj było lepiej niż gorzej

Reply:
Aha, no tak.

Normalnie x_dict_id oznacza id pozycji ze słownika x albo jakoś pokrewnego x, a tutaj oznacza id pozycji zupełnie dowolnej, jeżeli rozważamy tę tabelę w odosobnieniu. Więc może tu odpuścić tę konwencję _dict_id i zrobić id_value i position_id_value?

Zresztą konwencja _dict_id jest nadal zachowana poniekąd bo _dict_id jest w apiName atrybutu którego to jest wartość.

Reply:
generalnie prościej jest nadać jakąś nazwę, i najwyżej później ją zmienić, to nie jest trudne

trudne jest, jak nic się nie da ruszyć do przodu

Reply:
mówiłem, że ref mi pasuje, bo istotą tej wartości jest jej bycie referencją do jakiegoś obiektu, a nie samo bycie wartością

value.id_value aż takie lepsze niż value.ref_object_id?